apps/src/applab/applab.js

```diff
@@ -1651,3 +1653,18 @@ Applab.updateProperty = function (element, property, value) {
 Applab.isCrosshairAllowed = function () {
   return !Applab.isReadOnlyView && !Applab.isRunning();
 };
+
+Applab.showRateLimitAlert = function () {
+  // only show the alert once per session
+  if (hasSeenRateLimitAlert) {
+    return false;
+  }
+  hasSeenRateLimitAlert = true;
+
+  var alert = <div>The app has been reading or writing to the database too frequently.  Please try again a little later.  Let the developer know if you see this problem again.</div>;
```

Comment:
Changing still requires developer intervention (or at least someone able to make changes to our repo). In any case, I'm (a) not planning to merge before we have a better string and (b) should probably just move to i18n now regardless :)